parsec/core/logged_core.py

```diff
@@ -40,6 +45,55 @@
 
 logger = get_logger()
 
+FAILSAFE_PATTERN_FILTER = re.compile(
+    r"^\b$"
+)  # Matches nothing (https://stackoverflow.com/a/2302992/2846140)
+
+
+def _get_pattern_filter(pattern_filter_path: Path) -> Optional[Pattern]:
+    try:
+        data = pattern_filter_path.read_text()
+    except OSError as exc:
+        logger.warning(
+            f"Path to the file containing the filename patterns "
+            f"to ignore is not properly defined: {exc}"
+        )
+        return None
+    try:
+        regex = "|".join(
+            fnmatch.translate(line.strip())
+            for line in data.splitlines()
+            if line.strip() and not line.strip().startswith("#")
+        )
+    except ValueError as exc:
+        logger.warning(
+            f"Could not parse the file containing the filename patterns " f"to ignore: {exc}"
+        )
+        return None
+    try:
+        return re.compile(regex)
+    except re.error as exc:
+        logger.warning(
+            f"Could not compile the file containing the filename patterns "
+            f"to ignore into a regex pattern: {exc}"
+        )
+        return None
+
+
+def get_pattern_filter(pattern_filter_path: Optional[Path] = None) -> Pattern:
+    pattern = None
+    # Get the pattern from the path defined in the core config
+    if pattern_filter_path is not None:
+        pattern = _get_pattern_filter(pattern_filter_path)
+    # Default to the pattern from the ignore file in the core resources
+    if pattern is None:
+        with importlib_resources.path(core_resources, "default_pattern.ignore") as path:
```

Comment:
I don't think this is zipsafe, we should use `importlib.resources.read_text` instead

Reply:
It seems like it is:
> Exiting the context manager cleans up any temporary file created when the resource needs to be extracted from e.g. a zip file.

https://docs.python.org/3/library/importlib.html#importlib.resources.path

Reply:
That's pretty interesting !

(I'm still not a huge fan of thinking the underlying layers have to create a temp file on the FS so we can access it, but I don't have any serious reason to support this claim :smile: so let's leave this as it is !)